src/main/java/com/example/Session.java

```diff
@@ -0,0 +1,137 @@
+            Object value = resultSet.getObject(name, field.getType());
+            field.setAccessible(true);
+            field.set(entityInstance, value);
+        }
+        return entityInstance;
+    }
+
+    private static String getColumnName(Field field) {
+        return Optional.ofNullable(field.getAnnotation(Column.class))
+                .map(Column::value)
+                .orElseGet(field::getName);
+    }
+
+    private static <T> Field getIdField(Class<T> entityType) {
+        return Arrays.stream(entityType.getDeclaredFields())
+                .filter(f -> f.isAnnotationPresent(Id.class))
+                .findFirst()
+                .orElseThrow();
+    }
+
+    private static <T> String getTableName(Class<T> entityType) {
+        return Optional.ofNullable(entityType.getAnnotation(Table.class))
+                .map(Table::value)
+                .orElseGet(entityType::getName);
+    }
+
+    public void close() {
+        entitySnapshotMap.keySet()
+                .forEach(this::doEntityDirtyCheck);
+    }
+
+    @SneakyThrows
+    private void doEntityDirtyCheck(EntityKey k) {
+        Map<String, Object> columnsToUpdate = new HashMap<>();
+        if (entityCacheMap.containsKey(k)) {
+            Object entityInstance = entityCacheMap.get(k);
+            Field[] declaredFields = entityInstance.getClass().getDeclaredFields();
+            for (Field field : declaredFields) {
+                Object fieldValue = field.get(entityInstance);
+                Object fieldSnapshotValue = entitySnapshotMap.get(k).get(field.getName());
+                if (fieldValue != fieldSnapshotValue) {
+                    columnsToUpdate.put(getColumnName(field), fieldValue);
+                }
+            }
+            if (!columnsToUpdate.isEmpty()) {
+                update(entityInstance, columnsToUpdate);
+            }
+        }
+    }
+
+    @SneakyThrows
+    private void update(Object entityInstance, Map<String, Object> columnsToUpdate) {
+        try (Connection connection = dataSource.getConnection()) {
+            Class<?> entityType = entityInstance.getClass();
+            String tableName = getTableName(entityType);
+            String idColumnName = getColumnName(getIdField(entityType));
+            String setClauseString = mapToSetClauseString(columnsToUpdate);
+            String updateByIdQuery = "UPDATE %s SET %s WHERE %s = " .formatted(tableName, setClauseString, idColumnName);
+            try (PreparedStatement preparedStatement = connection.prepareStatement(updateByIdQuery)) {
+                preparedStatement.setObject(1, getIdField(entityType).get(entityInstance));
+                preparedStatement.executeUpdate();
+            }
+        }
+    }
+
+    private static String mapToSetClauseString(Map<String, Object> columnsToUpdate) {
+        return columnsToUpdate.entrySet()
+                .stream().map(e -> "%s = %s" .formatted(e.getKey(), e.getValue()))
```

Comment:
Typo here, the new value of column should be wrapped by `''` like `name = 'Simba'`